docs/requirements.md

```diff
@@ -2,88 +2,78 @@
 
 Running Codacy on a Kubernetes cluster requires the following:
 
--   A Kubernetes 1.14.\* or 1.15.\* cluster provisioned with the [required resources](#cluster-resource-requirements)
--   The [NGINX Ingress Controller](https://github.com/helm/charts/tree/master/stable/nginx-ingress) for Kubernetes
+-   A Kubernetes 1.14.\* or 1.15.\* cluster provisioned with the [required resources](#hardware-requirements)
+-   The [NGINX Ingress Controller](https://github.com/helm/charts/tree/master/stable/nginx-ingress) correctly set up in your cluster
 -   A [PostgreSQL server](#postgresql-server-setup) accessible from the Kubernetes cluster
 
-## Cluster resource requirements
+## Hardware requirements
 
-To have an enjoyable experience with Codacy, you should have the
-following calculations in mind when allocating resources for the
-installation and defining the number of concurrent analysis.
+The following high-level architecture is important in understanding how Codacy uses and allocates hardware resources.
 
-Without accounting for analysis ([next section](#analysis)),
-you should need at least the following resources:
+You can look at Codacy separately as the **"Platform"** and the **"Analysis"** cluster. The Platform contains the UI and other
+components important to treat and show results. The Analysis is the swarm of workers that run several linters for
+your project.
 
-```text
-CPU: 7 CPU
-Memory: 40 GB
-```
+**The resources needed for Codacy depend a lot on the rate of commits done by your team.**
 
-Check the
-[values-production.yaml](https://github.com/codacy/chart/blob/master/codacy/values-production.yaml)
-file to find a configuration reference that should work for you to run a
-"production-ready" version of Codacy.
+!["High Level Architecture"](<./images/High Level Architecture - Analysis II.svg> "High Level Architecture")
 
-### Analysis
+### Virtual CPUs and memory
 
-Each analysis runs a maximum number of 4 plugins in parallel (not configurable)
+Since all components are running on Kubernetes, you can increase the number of replicas in every deployment to give you more resilience and throughput, at a cost of increased resource usage. The following is a simplified overview of how to calculate resource allocation for the **"Platform"** and the **"Analysis"**:
 
-Note: All the following configurations are nested inside the `worker-manager.config`
-configuration object, but for simplicity, we decided to omit the full path.
+| Component                          | vCPU                    | Memory                   |
+| ---------------------------------- | ----------------------- | ------------------------ |
+| Platform (1 replica per component) | 4                       | 8                        |
+| Analysis Worker + 4 linters        | 5 (per Analysis Worker) | 10 (per Analysis Worker) |
```

Comment:
I think the "4 linters" part could be confusing for someone without the context of how our product works. People might confuse it with a specific set of linters instead of up-to 4 linters concurrently during an analysis.
Perhaps a note on this text could help?

Reply:
The idea was for the image to give this context.

I will try to write a bit about the "4 linters"